packages/datadog-plugin-redis/src/index.js

```diff
@@ -1,104 +1,101 @@
 'use strict'
 
-// TODO: always use uppercase for command names
-
-const tx = require('../../dd-trace/src/plugins/util/redis')
-
-function createWrapAddCommand (tracer, config) {
-  return function wrapAddCommand (addCommand) {
-    return function addCommandWithTrace (command) {
-      const name = command[0]
-      const args = command.slice(1)
-
-      if (!config.filter(name)) return addCommand.apply(this, arguments)
-
-      const scope = tracer.scope()
-      const span = startSpan(tracer, config, this, name, args)
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+const urlFilter = require('../../dd-trace/src/plugins/util/urlfilter')
+
+class RedisPlugin extends Plugin {
+  static get name () {
+    return 'redis'
+  }
 
-      return tx.wrap(span, scope.bind(addCommand, span).apply(this, arguments))
-    }
+  constructor (...args) {
+    super(...args)
+
+    this.addSub('apm:redis:command:start', ({ db, command, args, connectionOptions }) => {
+      this.config = normalizeConfig(this.config)
+      const store = storage.getStore()
+      const childOf = store ? store.span : store
+      const span = this.tracer.startSpan('redis.command', {
+        childOf,
+        tags: {
+          'span.kind': 'client',
+          'resource.name': command,
+          'span.type': 'redis',
+          'db.type': 'redis',
+          'db.name': db || '0',
+          'redis.raw_command': formatCommand(command, args)
+        }
+      })
+
+      span.setTag('service.name', this.config.service || `${span.context()._tags['service.name']}-redis`)
```

Comment:
This could be set above with the other tags instead of in a separate function call.